core/meterer/onchain_state.go

```diff
@@ -107,6 +109,33 @@ func (pcs *OnchainPaymentState) RefreshOnchainPaymentState(ctx context.Context,
 	}
 	// These parameters should be rarely updated, but we refresh them anyway
 	pcs.PaymentVaultParams = paymentVaultParams
+
```

Comment:
not necessarily this PR, but FYI. I ended up using atomic.Pointer to keep these "read-mostly" values

Reply:
cool, updated to use a pointer for this field